### PR TITLE
fix: dashboard asset balance

### PR DESCRIPTION
**Pull request by zhouhanseng**
Using the RPC of selected tab's chain rather than wallet current chain to fetch native token balance.

### CODE REVIEW

**Comment by AinuX** [comment]
Checked

File: packages/dashboard/src/pages/Wallets/hooks/useChainBalance.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import type { ChainId, ProviderType } from '@masknet/web3-shared-evm'
+import { useAsync } from 'react-use'
+import { Services } from '../../../API'
+
+export function useChainBalance(account: string, chainId: ChainId | null, providerType: ProviderType) {
```
Comment (guanbinrui) [inline packages/dashboard/src/pages/Wallets/hooks/useChainBalance.ts:5]: Move this hook to `packages/web3-shared/evm/hooks/useBalance`.

File: packages/dashboard/src/pages/Wallets/hooks/useChainBalanceList.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { NetworkType, ProviderType, getChainIdFromNetworkType, ChainId } from '@masknet/web3-shared-evm'
+import { useAsync } from 'react-use'
+import { Services } from '../../../API'
+
+export function useChainBalanceList(account: string, providerType: ProviderType, disabled: boolean) {
```
Comment (guanbinrui) [inline packages/dashboard/src/pages/Wallets/hooks/useChainBalanceList.ts:5]: Move this hook to `packages/web3-shared/evm/hooks/useBalance`.

File: packages/dashboard/src/pages/Wallets/hooks/useChainBalance.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+    return useAsync(
+        async () => (chainId ? Services.Ethereum.getBalance(account, { chainId }) : null),
```
Comment (guanbinrui) [inline packages/dashboard/src/pages/Wallets/hooks/useChainBalance.ts:7]: Would you please refactor to use `useWeb3()`?

File: packages/dashboard/src/pages/Wallets/hooks/useChainBalanceList.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    const chainIdList = Object.keys(NetworkType).map((x) => getChainIdFromNetworkType(x as NetworkType))
+    const allRequest = disabled
+        ? null
+        : chainIdList.map(async (chainId) => ({
+              balance: await Services.Ethereum.getBalance(account, { chainId }),
```
Comment (guanbinrui) [inline packages/dashboard/src/pages/Wallets/hooks/useChainBalanceList.ts:10]: Would you please refactor to use `useWeb3()`?

**Review by Jack-Works — DISMISSED** [review]

File: packages/web3-shared/evm/hooks/useBalance.ts [hunk 5]
```diff
@@ -1,8 +1,43 @@
...
+import { first } from 'lodash-es'
 
 /**
  * Get the current block number
  */
 export function useBalance() {
     return useWeb3StateContext().balance
 }
+
+export function useChainBalance(account: string, chainId: ChainId | null, providerType: ProviderType) {
```
Comment (guanbinrui) [inline packages/web3-shared/evm/hooks/useBalance.ts:16]:
```suggestion
export function useChainBalance(account: string, chainId: ChainId, providerType: ProviderType) {
```

File: packages/web3-shared/evm/hooks/useBalance.ts [hunk 6]
```diff
@@ -1,8 +1,43 @@
...
+    const web3 = useWeb3(true, chainId ?? ChainId.Mainnet)
+    return useAsync(async () => (chainId ? web3.eth.getBalance(account) : null), [account, chainId, providerType])
+}
+
+export function useChainBalanceList(account: string, providerType: ProviderType, disabled: boolean) {
```
Comment (guanbinrui) [inline packages/web3-shared/evm/hooks/useBalance.ts:21]: If the caller needs to depress the function, pass an empty account address to replace the `disabled` parameter.

File: packages/web3-shared/evm/hooks/useBalance.ts [hunk 7]
```diff
@@ -1,8 +1,43 @@
...
+    const web3 = useWeb3(true)
+    const chainIdList = Object.keys(NetworkType).map((x) => getChainIdFromNetworkType(x as NetworkType))
+    const allRequest = disabled
+        ? null
+        : chainIdList.map(async (chainId) => {
+              const { RPC } = getRPCConstants(chainId)
+              const providerURL = first(RPC)
+              if (providerURL) web3.setProvider(providerURL)
+              return {
+                  balance: await web3.eth.getBalance(account),
```
Comment (guanbinrui) [inline packages/web3-shared/evm/hooks/useBalance.ts:31]: If the `providerURL` does not exist, it should return `'0'` as the balance.

**Review by septs — APPROVED** [review]

**Review by guanbinrui — APPROVED** [review]

**Comment by AinuX** [comment]
Checked

**Pull request merged**